tests/check_doctest_names.py

```diff
@@ -0,0 +1,201 @@
+"""A helper script to check names in doctests.
+
+The problem is that pytest doctests (following the stdlib doctest
+module) see the module-global namespace. So when a doctest looks
+like this:
+
+    Examples
+    --------
+    >>> import numpy
+    >>> import vtk
+    >>> import pyvista
+    >>> offset = np.array([0, 9])
+    >>> cells = np.array([8, 0, 1, 2, 3, 4, 5, 6, 7, 8, 8, 9, 10, 11, 12, 13, 14, 15])
+    >>> cell_type = np.array([vtk.VTK_HEXAHEDRON, vtk.VTK_HEXAHEDRON], np.int8)
+
+there will be a ``NameError`` when the code block is copied into Python
+because the ``np`` name is undefined. However, pytest and sphinx test
+runs will not catch this, as the np name is typically also available in
+the global namespace of the module where the doctest resides.
+
+In order to fix this, we build a tree of pyvista's public and private
+API, using the standard-library doctest module as a doctest parser. We
+execute examples with a clean empty namespace to ensure that mistakes
+such as the above can be caught.
+
+Note that we don't try to verify that the actual results from each
+example are correct; that's still pytest's responsibility. As long
+as the examples run without error, this module will be happy.
+
+The implementation is not very robust or smart, it just gets the job
+done to find the rare name mistake in our examples.
+
+This module is executable. If you need off-screen plotting, set the
```

Comment:
> This module is executable

Is it?  Don't we need a shebang at L1?

Reply:
Or you just mean it's callable from python.  Is there a difference.  Correct me if I'm wrong.